pkg/sql/ttl/ttljob/ttljob_keytodatums_test.go

```diff
@@ -0,0 +1,33 @@
+package ttljob
+
+import (
+	"fmt"
+	"github.com/cockroachdb/cockroach/pkg/keys"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/types"
+	"github.com/cockroachdb/cockroach/pkg/util/encoding"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestKeyToDatums(t *testing.T) {
+
+	const hexKey = "2f54656e616e742f353535342f5461626c652f3132332f312f225c783163205c7865666660294c5c7866665c786265505c7862615c78646379475c7864665c78313522"
+	var bytes []byte
+	_, err := fmt.Sscanf(hexKey, "%x", &bytes)
+	require.NoError(t, err)
+	rKey := roachpb.RKey(bytes)
+	require.Equal(t, hexKey, fmt.Sprintf("%x", rKey))
```

Comment:
I reverted the fix for now until we can get this test failing to match the prod error.
For some reason I can't figure out how to create an `RKey` that results in `hexKey` when put through `fmt.Sprintf("%x", rKey)` though so I am having trouble reproducing it:
```
Expected :2f54656e616e742f353535342f5461626c652f3132332f312f225c783163205c7865666660294c5c7866665c786265505c7862615c78646379475c7864665c78313522
Actual   :222f54656e616e742f353535342f5461626c652f3132332f312f5c225c5c783163205c5c7865666660294c5c5c7866665c5c786265505c5c7862615c5c78646379475c5c7864665c5c7831355c2222
```

Reply:
The error output is not usable because of the bug that was fixed here https://github.com/cockroachdb/cockroach/pull/90727 so the test is not usable.